tasks/common.go

```diff
@@ -14,12 +14,15 @@ type Script struct {
 
 type Task interface {
 	GetTypeName() string
-	Validate() error
+	Validate(goos string) error
 	GetPath() string
 	GetRequirements() []string
 	GetCreatesFilesList() []string
 	GetOnlyIfCmds() []string
 	GetUnlessCmds() []string
+
+	GetTracker() (tracker *FieldStatusTracker)
```

Comment:
Can I suggest inverting this and having a `SetTracker` method.
You could then separate these into a separate `TaskWithTracker` interface. Builder could then at runtime check if `Task` is `TaskWithTracker` and only set the tracker if it is.
That way all other tasks wouldn't have to have support for tracker and change fields.

Reply:
see comment above.